### ISSUE TITLE
SPM fails to install additional modules correctly

### ISSUE THREAD

**Issue by gmacon** [saltstack/salt#44554]
### Description of Issue/Question

SPM install does not place the custom modules and states in the correct location (`/srv/spm/salt/_modules` and `/srv/spm/salt/_states`, respectively). It instead places them inside the formula directory. This is not the [documented behavior](https://docs.saltstack.com/en/latest/topics/spm/spm_formula.html#loader-modules), and it is a regression since 2016.11.

I suspect this may be related to #42646, but I didn't notice until now because we haven't deployed a new salt master or changed my custom states between the release of 2017.1.0 and now. I've confirmed that my upgraded salt master now contains copies in both the correct and incorrect locations, but the files in the correct location have not been touched since we upgraded to 2017.1 in July.

### Setup

Assume the existence of a SPM package repository containing a package `foo-formula` which contains a custom state (i.e. `foo-formula/_states/foo.py`)

### Steps to Reproduce Issue

```
spm install foo
ls /srv/spm/salt  # Note absence of _states
ls /srv/spm/salt/foo  # Note presence of _states
```

### Versions Report

```
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.5.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

**Comment by garethgreenaway** [comment]
@gmacon Thanks for the report!

**Comment by rallytime** [comment]
@gmacon This should be fixed with #50211
